lib/net/ssh/test/extensions.rb

```diff
@@ -156,6 +156,8 @@ def select_for_test(readers=nil, writers=nil, errors=nil, wait=nil)
               end
     
               raise "no readers were ready for reading, and none had any incoming packets" if processed == 0 && wait != 0
+
+              [[], [], []]
```

Comment:
This is returned because returning nil is interpreted as a timeout now (which is how the real IO.select behaves).